core/controllers/domain_objects_validator.py

```diff
@@ -96,3 +97,28 @@ def validate_collection_change(obj):
     # No explicit call to validate_dict method is necessary, because
     # CollectionChange calls validate method while initialization.
     collection_domain.CollectionChange(obj)
+
+
+def validate_task_entriy_for_improvements(task_entry):
+    # type: (Dict[String, Any]) -> None
+    """Validates collection change.
+
+    Args:
+        task_entry: dict. Data that needs to be validated.
+    """
+    # There is no validate method in domain layer.
+    entity_version = task_entry.get('entity_version', None)
+    if entity_version is None:
+        raise base.BaseHandler.InvalidInputException(
+            'No entity_version provided')
+    task_type = task_entry.get('task_type', None)
+    if task_type is None:
+        raise base.BaseHandler.InvalidInputException('No task_type provided')
+    target_id = task_entry.get('target_id', None)
+    if target_id is None:
+        raise base.BaseHandler.InvalidInputException('No target_id provided')
+    status = task_entry.get('status', None)
+    if status is None:
+        raise base.BaseHandler.InvalidInputException('No status provided')
+    # The issue_description is allowed to be None.
+    task_entry.get('issue_description', None)
```

Comment:
Is this line required?

Reply:
Removed, since it is not playing any role in validation.